docs/asciidoc/modules/ROOT/pages/misc/match-entities.adoc

```diff
@@ -0,0 +1,213 @@
+[[match-entities]]
+= Match entities
+:description: This section describes procedures and functions for matching entities.
+
+The library provides 2 procedure for matching entities:
+
+- apoc.node.match
+- apoc.rel.match
+
+[[matching-node]]
+== Matching nodes
+
+[.emphasis]
+"apoc.node.match(['Label'], identProps:{key:value, ...}, onMatchProps:{key:value,...})" - match nodes with dynamic labels, with support for setting properties on matched nodes
+
+=== Signature
+
+[source]
+----
+apoc.node.match(label :: LIST? OF STRING?, identProps :: MAP?, onMatchProps = {} :: MAP?) :: (node :: NODE?)
+----
+
+=== Input parameters
+[.procedures, opts=header]
+|===
+| Name | Type | Default | Description
+| labels | LIST? OF STRING? | null | The list of labels used for the generated MATCH statement. Passing `null` or an empty list will match a node without any constraints on labels. `null` or empty strings within the list are not supported.
+| identProps | MAP? | null | Properties that are used for MATCH statement.
+| onMatchProps | MAP? | {} | Properties that are set when a node is matched.
+|===
+
+=== Output parameters
+[.procedures, opts=header]
+|===
+| Name | Type
+|node|NODE?
+|===
+
+This procedure provides a more flexible and performant way of matching nodes than Cypher's https://neo4j.com/docs/cypher-manual/current/clauses/match/[`MATCH`^] clause.
+
+=== Usage Examples
+The example below shows equivalent ways of matching a node with the `Person` label, with a `name` property of "Billy Reviewer":
+
+// tag::tabs[]
+[.tabs]
+
+.apoc.node.match
+[source,cypher]
+----
+CALL apoc.node.match(
+  ["Person"],
+  {name: "Billy Reviewer"},
+  {lastSeen: datetime()}
+)
+YIELD node
+RETURN node;
+----
+
+.MATCH clause
+[source,cypher]
+----
+MATCH (node:Person {name: "Billy Reviewer"})
+SET node.lastSeen = datetime()
+RETURN node;
+----
+// end::tabs[]
+
+.Results
+[opts="header"]
+|===
+| node
+| (:Person {name: "Billy Reviewer", lastSeen: 2020-11-24T11:33:39.319Z})
+|===
+
+But this procedure is mostly useful for matching nodes that have dynamic labels or properties.
```

Comment:
I think it would be also great to mention its advantage in combination with apoc.json procedures, maybe adding another example as well (for rel as well)